### PR TITLE
Reoganize experiment and move time complexity

### PR DESCRIPTION
**Pull request by howjmay**

### CODE REVIEW

File: paper.tex [hunk 1]
```diff
@@ -269,11 +269,15 @@ \subsubsection{End-to-End-Encryption}
     \item Send the ciphertext along with signed MAC.
 \end{enumerate}
 
-The E2EE protocol presented in this paper combines a series of different cryptographic procedures. Each of them can be chosen on the basis of the working hardware and scenario by developers. On some certain devices, hardware acceleration would be supported. With hardware acceleration, the elapsed time could be dropped a step more. For example applying AES-NI can accelerate AES operations around 40\% on Intel devices.\cite{AES-NI-Acceleration}
+The E2EE protocol framework presented in this paper combines a series of different cryptographic procedures. Each of them can be chosen on the basis of the working hardware and scenario by developers. On some certain devices, hardware acceleration would be supported. With hardware acceleration, the elapsed time could be dropped a step more. For example applying AES-NI can accelerate AES operations around 40\% on Intel devices.\cite{AES-NI-Acceleration}
```
Comment (jserv) [inline paper.tex]: E2EE protocol 似乎不足以說是 "framework"，畢竟只是個 fastpath

Comment (howjmay) [reply paper.tex]: 收到

File: paper.tex [hunk 2]
```diff
@@ -269,11 +269,15 @@ \subsubsection{End-to-End-Encryption}
...
+
+\subsubsection{Time Complexity}
+In this paper, we choose SHA-256 for HMAC, AES-CBC for symmetric encryption and ECDSA P-256 for digital signature. In this E2EE protocol, ECDSA would take constant time, since the string needed to be signed is in fixed length. HMAC and AES-CBC are the procedures that cost most of the elapsed time. Time complexity of HMAC with SHA-256 is $O(n)$.\cite{hmac_time_complexity} Block cipher takes constant time for a single block which cause the time complexity is $O(n)$, too. Therefore, the E2EE protocol framework takes only linear time versus arbitrary string length.
```
Comment (jkrvivian) [inline paper.tex]: Ditto, `framework`.

File: paper.tex [hunk 3]
```diff
@@ -269,11 +269,15 @@ \subsubsection{End-to-End-Encryption}
...
 
 \subsubsection{Issues in End-to-End-Encryption}
 E2EE helps a lot for low-level edge device utilizing blockchain service. It dramatically reduces execution time encryption takes. However, there are some challenges in the E2EE process. Tampering the messages which have conducted E2EE would be a challenging mission. However, spamming would be the critical issue participants may meet under this architecture. It is risky for an edge device connects to a tangle-accelerator which is operated by unknown third-party. Sharing Channel Chain ownership with an unknown third-party would allow them to spam on the MAM Channel Chain. Spam would cause subscribers wasting plenty of time on decrypting useless messages. To prevent edge devices from the annoying attack, connecting to an known, trustworthy tangle-accelerator would be the easiest solution.
 
+
```
Comment (jkrvivian) [inline paper.tex]: Remove the extra blank line.

File: paper.tex [hunk 4]
```diff
@@ -374,15 +378,15 @@ \subsection{MAM vs. the delegated MAM}
...
+Applying remote PoW greatly reduce the elapsed time. Channel creation is the next primary barrier for low-level devices to broadcast MAM message. With employing delegated MAM and E2EE, MAM channel creation would be delegated by tangle-accelerator which dramatically enhances the efficiency and makes MAM available for low-level devices. What follows is the performance comparison between MAM and delegated MAM for different payload sizes. Elapsed time for internet communication and PoW are not included, because the processes after sending message to tangle-accelerator are the common steps. Excluding the common steps allows us to compare the performance between MAM and delegated MAM more clearly. In addition, MAM Channel creation and E2EE asymmetric key pair generation are not included, too, because the two operations would seldom be executed. In the E2EE protocol framework, we choose SHA-256 for HMAC, AES-CBC for symmetric encryption and Curve P-256 for ECDSA, respectively.
```
Comment (jkrvivian) [inline paper.tex]: * Applying remote PoW greatly reduce`s` the elapsed time.
* ...to broadcast MAM message`s`
* ...be delegated `to` `T`angle-accelerator...
* sending message`s` to `T`angle-accelerator..
* In addition, MAM `c`hannel...
* Fix framework, here.

File: paper.tex [hunk 5]
```diff
@@ -406,15 +410,12 @@ \subsubsection{Experiment}
...
-With given E2EE protocol framework, ECDSA would take constant time under our situation, since the string needed to be signed is in fixed length. HMAC and AES-CBC are the procedures that cost most of the elapsed time. Time complexity of HMAC with SHA-256 is $O(n)$.\cite{hmac_time_complexity} Block cipher takes constant time for a single block which cause the time complexity is $O(n)$, too. Therefore, the E2EE protocol framework takes only linear time versus arbitrary string length. Applying mature cryptographic functions (e.g. SHA-256, AES-CBC, etc), ensures gentle slope as Eq.~\ref{eq:lin_e2ee} shows.
+Table~\ref{tab:mam_vs_e2ee} is the experiment result of the two methods. Eq.~\ref{lin_mam},~\ref{lin_e2ee} are the linear equations for the experiment data of MAM and E2EE respectively. The result reveals elapsed time of E2EE is obviously less than elapsing time of MAM which shows the E2EE process presented in the paper is a much more appropriate solution for resource-constrained systems to send authenticated streaming data. The slope of the result of E2EE is much more flat than the result of MAM which ensures E2EE can consistently perform in an acceptable time range.
```
Comment (jkrvivian) [inline paper.tex:417]: * elapsed time vs. elapsing time?

File: paper.tex (hunk 1 above)
Comment (howjmay) [reply paper.tex]: 我把 framewrork 拿掉變成 E2EE protocol

File: paper.tex (hunk 5 above)
Comment (jserv) [reply paper.tex:417]: Use elapsed time.

**Pull request merged**